templates/terraform/google-sa/main.tf

```diff
@@ -0,0 +1,44 @@
+# a service account
+resource google_service_account sa {
+  provider = google.target
+  account_id = var.account_id
+  display_name = var.display_name
+}
+
+# a key associated with the service account
+resource google_service_account_key sa_key {
+  service_account_id = google_service_account.sa.name
+}
+
+# a vault secret containing the JSON key for the service account
+resource vault_generic_secret sa_secret {
+  path = var.vault_path
+  data_json = <<EOT
+{
+  "sa_key": ${jsonencode(base64decode(google_service_account_key.sa_key.private_key))}
```

Comment:
I'm not a huge fan of `sa_key`. Could we just do `key` if there isn't a useful more-specific name?